File: client/minimalist/theme.py
"""Dark theme color definitions inspired by Linear's design language."""

import wx

# Dark theme color palette
DARK = {
    # Backgrounds
    'bg_primary': '#0D0D0D',
    'bg_secondary': '#141414',
    'bg_surface': '#1A1A1A',
    'bg_hover': '#222222',
    'bg_selected': '#2A2A2A',
    'bg_input': '#1E1E1E',

    # Text
    'text_primary': '#EBEBEB',
    'text_secondary': '#8A8A8A',
    'text_muted': '#5C5C5C',
    'text_accent': '#7C8AFF',

    # Borders
    'border_primary': '#2A2A2A',
    'border_hover': '#3A3A3A',

    # Accents
    'accent_blue': '#5E6AD2',
    'accent_purple': '#7C5CFC',
    'accent_green': '#10B981',
    'accent_yellow': '#F59E0B',
    'accent_red': '#EF4444',
    'accent_orange': '#F97316',
}

# Status category colors
STATUS_COLORS = {
    'backlog': '#6B7280',
    'unstarted': '#6B7280',
    'started': '#F59E0B',
    'completed': '#10B981',
    'cancelled': '#EF4444',
}

# Priority colors and labels
PRIORITY = {
    0: {'label': 'No Priority', 'color': '#6B7280', 'icon': '---'},
    1: {'label': 'Urgent', 'color': '#EF4444', 'icon': '!!!'},
    2: {'label': 'High', 'color': '#F97316', 'icon': '!! '},
    3: {'label': 'Medium', 'color': '#F59E0B', 'icon': '!  '},
    4: {'label': 'Low', 'color': '#6B7280', 'icon': '.  '},
}

# Estimate options (Fibonacci scale)
ESTIMATES = [1, 2, 3, 5, 8, 13, 21]


def hex_to_wx(hex_str):
    """Convert hex color string to wx.Colour."""
    hex_str = hex_str.lstrip('#')
    r = int(hex_str[0:2], 16)
    g = int(hex_str[2:4], 16)
    b = int(hex_str[4:6], 16)
    return wx.Colour(r, g, b)


def apply_dark_theme(widget, bg_key='bg_primary', fg_key='text_primary'):
    """Apply dark theme colors to a widget."""
    widget.SetBackgroundColour(hex_to_wx(DARK[bg_key]))
    widget.SetForegroundColour(hex_to_wx(DARK[fg_key]))

File: ios/Minimalist/Theme/Theme.swift
import SwiftUI

/// Dark theme matching the desktop client's color palette.
enum Theme {
    // Backgrounds
    static let bgPrimary = Color(hex: "0D0D0D")
    static let bgSecondary = Color(hex: "141414")
    static let bgSurface = Color(hex: "1A1A1A")
    static let bgHover = Color(hex: "222222")
    static let bgSelected = Color(hex: "2A2A2A")

    // Text
    static let textPrimary = Color(hex: "EBEBEB")
    static let textSecondary = Color(hex: "8A8A8A")
    static let textMuted = Color(hex: "5C5C5C")

    // Accents
    static let accentPurple = Color(hex: "7C5CFC")
    static let accentBlue = Color(hex: "5E6AD2")
    static let accentGreen = Color(hex: "10B981")
    static let accentYellow = Color(hex: "F59E0B")
    static let accentRed = Color(hex: "EF4444")
    static let accentOrange = Color(hex: "F97316")

    // Borders
    static let borderPrimary = Color(hex: "2A2A2A")

    // Status category colors
    static func statusColor(_ category: String) -> Color {
        switch category {
        case "backlog", "unstarted": return Color(hex: "6B7280")
        case "started": return accentYellow
        case "completed": return accentGreen
        case "cancelled": return accentRed
        default: return Color(hex: "6B7280")
        }
    }

    // Priority colors
    static func priorityColor(_ priority: Int) -> Color {
        switch priority {
        case 1: return accentRed
        case 2: return accentOrange
        case 3: return accentYellow
        case 4: return Color(hex: "6B7280")
        default: return Color(hex: "6B7280")
        }
    }

    static func priorityLabel(_ priority: Int) -> String {
        switch priority {
        case 1: return "Urgent"
        case 2: return "High"
        case 3: return "Medium"
        case 4: return "Low"
        default: return "No Priority"
        }
    }
}

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let r, g, b: UInt64
        (r, g, b) = ((int >> 16) & 0xFF, (int >> 8) & 0xFF, int & 0xFF)
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: 1
        )
    }
}
